### PR TITLE
Cert personal insights

### PR DESCRIPTION
**Pull request by TheMOHD3886**
## Description

Created functionality for the personal insights pane, which has also been tested. Also done minor changes here and there.

## Changes

- Welcome message now uses the user's name
- Counts total amount of courses
  - Total amount of published courses
  - Total amount of draft courses
  - Total amount of hidden courses
- Shows the average star rating
- Total amount of subscribers

## Related Issues

None.

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

image

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: dev
Commit id: a1834a91346af44d16203f89061f7aa8a504f01f

## Notes for Reviewers

None.

### CODE REVIEW

File: app/__tests__/src/components/Courses/__snapshots__/PersonalInsights.test.js.snap
Comment (magnemo) [inline app/__tests__/src/components/Courses/__snapshots__/PersonalInsights.test.js.snap:1]: Kan vi lave test uden snapshorts, de er træls

File: app/__tests__/src/pages/__snapshots__/Courses.test.js.snap
Comment (magnemo) [inline app/__tests__/src/pages/__snapshots__/Courses.test.js.snap:1]: Kan vi lave test uden snapshorts, de er træls

File: app/src/components/Courses/PersonalInsights.tsx [hunk 1]
```diff
@@ -19,52 +25,89 @@ type PersonalInsightsProps = {
 const PersonalInsights = (props: PersonalInsightsProps) => {
   const courses = props.courses ?? [];
 
+  // Gets the total number of subscribers from all courses written by the content creator
+  function getAllSubscribers() {
```
Comment (magnemo) [inline app/src/components/Courses/PersonalInsights.tsx:29]: Hvorfor er getAllSubscribers, getNumberOfCourses, getAverageRating, getCourseCountWithStatus ikke funktioner i userinfo

File: app/src/helpers/userInfo.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+type userInfo = {
+  name: string,
+  email: string,
+  id: string,
+}
+
```
Comment (magnemo) [inline app/src/helpers/userInfo.ts:6]: tilføje getAllSubscribers, getNumberOfCourses, getAverageRating, getCourseCountWithStatus, så de kan genbruges andre steder

File: app/__tests__/src/pages/Courses.test.js
Comment (Viigg0) [inline app/__tests__/src/pages/Courses.test.js:1]: Maybe add comment to clarify the tests purpose

**Review by magnemo — CHANGES_REQUESTED** [review]

File: app/src/components/Courses/PersonalInsights.tsx [hunk 3]
```diff
@@ -19,52 +25,89 @@ type PersonalInsightsProps = {
...
-            <p>Total cursos</p>
-            <p className='font-bold'>8</p>
-          </div>
-          <div>
-            {/* Total students */}
-            <p>Total alunos</p>
-            <p className='font-bold'>167</p>
-          </div>
-          <div>
-            {/* Total certificates issued */}
```
Comment (Viigg0) [inline app/src/components/Courses/PersonalInsights.tsx:44]: Is total certificates not needed or will it just be implemented another time

**Comment by magnemo** [comment]
PLZ vent med at puch til dev før VT ændringerne er inde, just in case det fucker med noget

**Review by Viigg0 — COMMENTED** [review]
Looks good only found 2 things!

File: app/src/components/Courses/PersonalInsights.tsx (hunk 1 above)
Comment (Havkost) [reply app/src/components/Courses/PersonalInsights.tsx:29]: Moved them into a courses.ts helper. Doesn't really make sense to have them together with userInfo 👍

File: app/__tests__/src/components/Courses/__snapshots__/PersonalInsights.test.js.snap
Comment (Havkost) [reply app/__tests__/src/components/Courses/__snapshots__/PersonalInsights.test.js.snap:1]: Enig i at vi skal ændre vores frontend testing til at være unit tests af funktioner, og user journey testing af mere makroskopiske flows. Det skal ændres i mange test filer though, ikke bare her, så jeg tænker ikke det giver mening at refactor i det her pull request.

File: app/src/components/Courses/PersonalInsights.tsx (hunk 3 above)
Comment (Havkost) [reply app/src/components/Courses/PersonalInsights.tsx:44]: They will be changed, instead we'll show the percentage of students that have completed the user's courses 👍

File: app/__tests__/src/pages/Courses.test.js
Comment (Havkost) [reply app/__tests__/src/pages/Courses.test.js:1]: To me the descriptions for the individual tests are enough, what specifically would you like elaborated? - note the only change to this specific test file is mocking the userInfo helpers, not the tests themselves.

File: app/__tests__/src/components/Courses/__snapshots__/PersonalInsights.test.js.snap
Comment (magnemo) [reply app/__tests__/src/components/Courses/__snapshots__/PersonalInsights.test.js.snap:1]: fair

File: app/src/components/Courses/PersonalInsights.tsx (hunk 1 above)
Comment (magnemo) [reply app/src/components/Courses/PersonalInsights.tsx]: Sounds good

**Review by JackSMunn — APPROVED** [review]
Seems good to me, however i did not test locally, so you need more approves.

**Review by MortenMunk — APPROVED** [review]
Looks fine to me 👍🥇

**Review by magnemo — APPROVED** [review]
cool code

**Pull request merged**